Create a reasonable dense mapping pipeline for HoloLens

michalpolic opened the pull request:
Add two nodes into Meshroom: HololensPointcloudComposer, KeyframeSelector
Update the documentation and assertion of the codes.

Commit:
Initial commit.

Commit:
Better IPcamera, improved viz flags and performance, remove overlaps from GT, fixed typos, add comments

Commit:
New yfcc100m pairs file and readme entry, add fixed EXIF rotations, relative paths in pairs files, display thresholds, general cleanup

sarlinpe committed:
Do not store attention masks by default to save memory

michalpolic committed:
Update gitignore

michalpolic committed:
Add initialization of code modules

michalpolic committed:
Add HoloLens pointcloud composer

Pointcloud composer does load, decode and transfrom to one coordinate
system the depthmaps.

michalpolic committed:
Update dense pointcloud composer output path

michalpolic committed:
Add keyframe selector node

Add the node for selecting the keyframes based on its blur and distance
from previous keyframe. Update the assertions and documentation.

michalpolic committed:
Commit changes to server

This commit gather changes of paths, naming, docs, add docker support,
and share the initial commit of HoloLensMatcher.py.

michalpolic committed:
Add node for dense pointcloud filtering

Commit:
Merge modification in server and origin

Commit:
Merge branch 'dev' of github.com:michalpolic/hololens_mapper into dev

Commit:
Add Munich benchmark dataset

zderaann commented on third_party/meshroom/meshroom/nodes/ARTwin/HololensPointcloudComposer.py:
```diff
@@ -0,0 +1,94 @@
+            uid=[0],
+        ),
+        desc.ChoiceParam(
+            name='verboseLevel',
```
There are different types of quotation marks used for strings ', ''' and ". It would be best for consistency reasons to use only one type where it is possible.

zderaann commented on third_party/meshroom/meshroom/nodes/ARTwin/HololensPointcloudComposer.py:
```diff
@@ -0,0 +1,94 @@
+
+    inputs = [
+        desc.File(
+            name="recordingDir",
```
It's just a stylistic note and maybe it cannot be done here but it would be better to have consistency in spaces around equal sign.

zderaann commented on third_party/meshroom/meshroom/nodes/ARTwin/HololensPointcloudComposer.py:
```diff
@@ -0,0 +1,94 @@
+                chunk.logger.warning('Missing depth decoding file (uvdata.txt)')
+                return
+            if not chunk.node.output.value:
+                return
```
I am not sure what's the problem here. If I had to guess it would be probably because the user didn't input the output folder, it would be nice to let the user know, why it returned without doing anything (could we add some log message like in the previous two if conditions?)

zderaann commented on third_party/meshroom/meshroom/nodes/ARTwin/HololensPointcloudComposer.py:
```diff
@@ -0,0 +1,94 @@
+            holo_io = HoloIO()
+            
+            chunk.logger.info('Reading HoloLens depthmaps.')
+            holo_xyz = holo_io.read_dense_pointcloud(chunk.node.recordingDir.value + "/long_throw_depth", 
```
We have here both camel case and snake case notation (processChunk vs read_dense_pointcloud, holo_xyz vs recordingDir) which could be confusing (perhaps we should decide which notation to use or use one for methods and the other for variables, etc.. ).

zderaann commented on third_party/meshroom/meshroom/nodes/ARTwin/KeyframeSelector.py:
```diff
@@ -0,0 +1,105 @@
+from __future__ import print_function
```
In this file there are the same notes as in HololensPointCloudComposer, but these are just minor remarks, overall the code is very well readable and easily understandable.

zderaann reviewed (commented):
I added a few remarks but overall the code is very well understandable and readable :)

Commit:
Update keyframe selector

Commit:
Add script for visibility estimation

VisibilityEstimation.py estimates the visibility for dense pointcloud
and (sub)set of cameras. The slowest parts
 are rewritten to Cpp and wrapped by pybind11 into Python.

Commit:
Add support of AliceVision containers

This is initial commit which has to be tested on Windows. This commit
add node to init the enviroment before running Meshroom nodes.

Commit:
Update gitignore

Commit:
Update & fix of VisibilityEstimation executable

The commint simplify codes, update paths, remove unnescesarry spaces in
output sfm.json file, remove bug in initialization of cpp array and
add filter of 3D points behind the camera.

michalpolic committed:
Remove old files

michalpolic committed:
Squashed 'third_party/meshroom/' changes from 5bdbeaf0..c3cfa21b

c3cfa21b Merge pull request #1545 from alicevision/dev/hideWidgets
36896c12 Merge pull request #1553 from alicevision/dev/batchCompute
e5b82cb1 [bin] meshroom_batch: Add verbose option
ae0c65f5 [core] do not use the task manager in batch command line tools
dedc5b44 [ui] 3DViewer: use a Loader to completely disable the 3D environment
520efa98 Merge pull request #1547 from asoftbird/develop
aa14fbc3 Merge pull request #1551 from alicevision/fix/uvCenterOffset
cd15a923 [ui] fix uvCenterOffset
0d61ebd5 Update StatViewer.qml
0f951889 [ui] widgets visibility options
1c66c6d9 Merge pull request #1537 from alicevision/dox/bibtexfix
a5676536 [doc] add HAL link to the paper
f45174fb [doc] fix the bibtex
95c273f4 [doc] fix latex citation
5e609747 Merge pull request #1349 from N0Ls/dev/panoramaViewer
8112cdbd [ui] qml: remove trailing log
0cb8171c [multi] Pass the right sfm node according to the type of viewer
38cd58f0 [Panorama Viewer] Adjust comments and increments
ba42c5e8 [Panorama Viewer] Update condition for selected view change
f25b5fe7 [Panorama Viewer] Change the method to get the fisheye circle parameters
3fad2faa [multi] Add new active node category and pass id view to disto viewer
a8936c46 [ui] Change selected view when hovered inside pano viewer
4d169f2b [Panorama Viewer] allow to highlight multiple images
a8109444 [Panorama Viewer] Change default and max values for subdivisions
b24b59d1 [Panorama Viewer] Update status of the fisheye circle button
7e65c91b [Panorama Viewer] Add fisheye support for viewer
67fab198 [Viewer2D] ImageViewer: fix MSfmData Qt Binding
db299d49 [python] Remove trailing whitespaces
e43e78d1 [Panorama Viewer] Add undo group for panorama modifications
77b23aef [Panorama Viewer] Remove debugs
9ffc2f3b [Panorama Viewer] rename signal
c1e57f6e [Panorama Viewer] remove unused variables
fb70a355 [Panorama Viewer] Add String to Url conversion for image path
c4cb893b [Panorama Viewer] Use msfmData to load images
223a3897 [Panorama Viewer] Use msfmData from PanoramaViewer (WIP)
b6be11f3 [Panorama Viewer] Add pitch and roll properties
67995989 [wip] Use qt binding on angles for edition
9b753cc3 [Python Graph] Expose modification group methods
262e42a1 [Panorama Viewer] Force display of sfmDataTransform sfm in pano viewer
6cd64b96 [Panorama Viewer] Change sfm filepath handling in mSfMData object
f37166da [Panorama Viewer] Simplify use of mSfMData object for pano viewer
188d9145 [Panorama Viewer] Add roll edit with mouse to panorama viewer
4af4faff [Panorama Viewer] rename surface rotation functions
c88f1e84 [Fix] Fix fisheye center attribute setting
367f763e [Panorama Viewer] Improve undo steps for manual changes
b998b493 [Panorama Viewer] Allow undo on manual changes
28e90c04 [Panorama Viewer] Add roll axis preview
c06273d2 [Panorama Viewer] clean up panorama code
4aa5151a [Panorama Viewer] Fix principal point display
c6b2e2d0 [Panorama Viewer] use degrees rotate function
7c8431bc [Panorama Viewer] apply node values on Initialization
86a09530 [Panorama Viewer] isComputed does not affect active property
930d885c [Panorama Viewer] use msfmDataLoader with Panorama and Disto Viewer
cd3d3cc1 [Panorama Viewer] fix msfmData status bug
2635e51f [wip] Change sfmData handling in panorama viewer
20b20845 [Panorama Viewer] Uncheck panorama viewer button when disabled
88e42de2 [Panorama Viewer] Bind sfmRequired property
63f00899 [Panorama Viewer] Load images two by two
b9db4e80 [Panorama Viewer] Change the state of hover bool on viewer set
5af84f51 [Panorama Viewer] change default subdivisions value
843f1474 [Panorama Viewer] Change active conditions for Float Image Viewer
cfd03bb2 [Panoram Viewer] Avoid problems with null state for image container
73999c43 [Panorama Viewer] Keep HDR viewer checked between scenes
1dfe96c7 [Panorama Viewer] Load images one by one
4a7912c0 [Panorama Viewer] Disable color picker inside HdrToolbar for Panorama
6534fc11 [Panorama Viewer] Disable display feature button when pano is enabled
4458a419 [Panorama Viewer] Link rgba channel dropdown inside pano viewer
25f6b57b [Panorama Viewer] Use qProperty for sfmPath and downscale
7d5c65e6 [Panorama] Retrieve downscale from qInvokable
7e1bff04 [Float Image Viewer] fix viewer condition issues due to last commit
b81e6852 [Float Image Viewer] Change viewer type with switch case on EViewerType
266aee8d [Panorama Viewer] fix surface grid bug when downscale
baf02add [Panorama Viewer] Change subdivisions with qProperty
72ad12d1 [Lens Distortion Viewer] Change grid opacity with qProperty
dc7ee20b [Lens Distortion Viewer] Remove control points on distortion viewer
fdbc5fa9 [Panorama Viewer] fix gridDisplay bug when changing viewer
4937af5c [Panorama Viewer] Change default values for disto viewer
b8bf027d [Panorama Viewer] change viewer type with qt property
157799c2 [Panorama Viewer] Rotate from the surface
7ec293a4 [Panorama Viewer] Call SfmPath methods from surface
443b3d29 [Panorama Viewer] Call Mouse Over methods from surface
6fb977f2 [Panorama Viewer] add surface display grid property
e642b970 [PanoramaViewer] Change function calls to adapt to surface change in cpp
634d8a20 [PanoramaViewer] Move vertex logic to surface
cf01a2d8 [PanoramaViewer] Refactor downscale code logic
a4a86e96 [ui] PanoViewer : allow gamma and gain editing
8f250863 [Panorama Viewer] Move inital state code block
ae511e0e [Panorama Viewer] Change syntax logic for more consistency
3ea4e277 Change colorGrid with Surface Q_Property
efb570c3 [ui] Viewer2D: enabled status on panorama and lens buttons
7c5752e7 [ui] Viewer2D: metadata can come from sfm or the image itself
8877c86d [Panorama Viewer] Fix enabled pano viewer button at start
9d0c4617 [Panorama Viewer] fix highlight bug
b36f5f3f [node] SfmTransform - manual transformation method by default
8c14387b [Panorama Viewer] Fix HDR display bug on new pipeline action
4b2b071f [Panorama Viewer] Add rotation to float image viewer
e432a1b7 [Panorama Viewer] fix hightlight z order bug
f8a1e006 [Panorama Viewer] Fix crash on new pipeline action
cf6d8215 [Panorama Viewer] new algo for hightlight images
423e82e2 [Panorama Viewer] auto fit panorama when ready
dc693040 [Panorama Viewer] HDR Color picker fix
05d50c70 [Panorama Viewer] Loading indicator added
9decce4e [Panorama Viewer] fix ram commit issues
8fa7f18c [Panorama Viewer] ensure it fits in RAM memory
1d1d0bb6 [Panorama Viewer] change max subdivisions value
9a6a8e19 [Panorama Viewer] change visibility panorama button condition
2e1c80b8 [Panorama Viewer] change pitch and yaw from SfMTransform Node
733362fe [Panorama Viewer] Pass Euler angle to sfm node
ed853d56 [Panorama Viewer] display hand cursor only if pano is editable
5fba58c1 [Panorama Viewer] correct ratio for panorama
dadb8754 [Panorama Viewer] add mouse speed multiplier
b66a19d0 [Panorama Viewer] change cursor shape when edit panorama
5b5f725f [Panorama Viewer] refactoring panorama viewer
8216fdd2 [Panorama Viewer] fix drag movement problem
7e22f672 [Panorama Viewer] downscale spinbox works
8ee2daa5 [Panorama Viewer] fix highlight bug
a2c658a0 [Panorama Viewer] update subdivisions + speed Y Axis
a8f6c280 [Panorama Viewer] change editing speed on scroll
2a1e6811 [Panorama Viewer] add grid display button function
fcdc6cc8 [Panorama Viewer] add subdivisions slider
e8534534 [Panorama Viewer] draw static grid inside panorama viewer
1f0f2e33 [Panorama Viewer] add button interactivity
1b5d2a19 [Panorama Viewer] send value from toolbar to viewer
508775c0 [Panorama Viewer] add Panorama Viewer toolbar
0a2fc484 [Panorama Viewer] rotate panorama with mouse position
cb4c56bc [Panorama Viewer] re order z index when highlight
627278b4 [Panorama Viewer] Swap source and panoViewer in Qt Bindings
16c62441 [Panorama Viewer] Highlight onMouseOver
5755b351 [Panorama Viewer] add mouse events for panorama editing
f06f5fcb [Panorama Viewer] fix button conflict problem
e2d8b61c [Panorama Viewer] Pano button link with C++
2a96a247 [Panorama Viewer] Setting subdivisions for panorama viewer
b478ca46 [Panorama Viewer] send IdView to C++
c9f965c0 [Panorama Viewer] Disable button when no SfM node is computed
edfcb2bb [Panorama Viewer] make repeater working with c++
d8018191 wip [PanoramaViewer] make repeater load different file paths
fdfe5d9d [Panorama Viewer] add Repeater
a33810f5 [Panorama Viewer] Add FloatImageViewer inside PanoramaViewer
874aff45 [PanoramaViewer] fix channelMode issue
660c3102 [PanoramaViewer] fix loader problem
0d45fc0b [PanoramaViewer] temporary change in gitignore
3ee49d11 [PanoramaViewer] add Panorama Viewer button and loader
77f6cdee [PanoramaViewer] add viewer files
aa223f22 wip [PanoramaViewer] add viewer files
fbaf76f5 [Distortion Viewer] sfmPath empty by default
e42c9577 [Distortion Viewer] change subdivisions stepsize to 2
d4268a83 [Distortion Viewer] fix function name typo
6f002d9a [Distortion Viewer] refactoring lens distortion related files
bab5e71a [Distortion Viewer] fix null item problem
7a870ab3 [Distortion Viewer] Disable Disto button when there is no sfm data
43e730a3 [Distortion Viewer] fix subdivisions display problem
b1a33137 [Distortion Viewer] Principal Point + dynamic sfm
11e0bd54 [Distortion Viewer] dynamic sfm data
01aa1cd9 [Distortion Viewer] update sfmData path
097eb66a [Distortion Viewer] Switch between viewer modes
fddbc885 [Lens Distortion] Display points only in distortion viewer
9e7e7320 [Lens Distortion] slider subs default value
c3e504ef [Lens Distortion] add control points display and features
b64b18c1 [Lens Distortion] subdivisions slider
ee563e22 [Lens Distortion] grid display on off
1225cf17 [Lens Distortion] Display grid only in Distortion Mode
29b7e808 [Lens Distortion] Update UI
c9f0ab0d [Lens Distortion] fix translation mistake
9600998b [LensDistorsion] default points HDR Viewer
ceb08c2e [Lens Distorsion] typo fix
cd97601f [Lens Distorsion] Renaming PanoramaViewer into Lens Distorsion
5e7f4faa [PanoramaViewer] add grid color change
1fc7cb78 [PanoramaViewer] add control points display buttons
8dc30a9d [PanoramaViewer] add grid opacity slider
41a5e5c1 [PanoramaViewer] grid not displayed by default
765dfce3 [PanoramaViewer] add grid display button
6a152ca6 [PanoramaViewer] add panorama toolbar
48d118e4 [PanoramaViewer] zoom fix
05ded4a8 [PanoramaViewer] zoom issue
ccd26f84 [PanoramaViewer] add grid display by default
e23b20cd wip [panoramaViewer] add viewer files
a9abd290 Merge pull request #1489 from remmel/dev_nocudapipeline
2a6bca30 Merge pull request #1521 from alicevision/fix/batch
d30571ce Draft Reconstruction pipeline
53011046 [ui] ImageGallery: call makeCurrentItemVisible only once
8beb8219 [nodes] CameraInit: use os.path.join
f44e8349 [core] fix batch
1a9692d4 Merge pull request #1524 from alicevision/cleanup/meshIO
61bc5992 [nodes] Mesh*: use file extension to choose the file format
9724e7e3 Merge pull request #1508 from Thomas-Zorroche/dev/normalMaps
f771dda9 [nodes] ConvertMesh: output file type is no more a command line argument
dcb984bf [nodes] Texturing: add upgrade code for colorMappingFileType
290e0278 [core] fix: filter attributes after the node upgrade
45bed696 [nodes] change default mesh format to obj
ef11acb4 [nodes] add outputMeshFileType parameter for meshMasking
c919e815 [Viewer3D] add .fbx and .gltf to supported extensions
fa0509ec [nodes] display material output only for .obj
dc1ca91a [core] node: warning on lambda evaluation
93f8595b [nodes] upgrade Texturing version to 6.0
4346e55d [nodes] separate choice parameter for bump mapping file type
b21ac279 [nodes] add lambda for Textures output value
f9f88c99 [nodes] change texture mapping parameters
440ec43a [texturing] fix export filename
295dee5d [nodes] ConvertMesh: add out mesh file type parameter
f26aea65 [nodes] outputMeshFileType: rename gltf2 into gltf
173be9f1 [nodes] add output mesh file type for Meshing and Mesh Filtering
2aa0bb12 [nodes] add mesh file type option
45c04113 [nodes] add group attribute for textures parameters
979fbde2 [nodes] export heightmaps only in .exr
9be05709 [nodes] add empty value for file type
cde2c7b3 Merge pull request #1500 from N0Ls/dev/ImageGalleryFilter
da1f2396 [ui] ImageGallery: fix sort and filtering
d3559bd5 [ui] fix image selection in ImageGallery
af4dae9a Merge branch 'dev/ImageGalleryFilter' of https://github.com/N0Ls/meshroom into N0Ls-dev/ImageGalleryFilter
b1c421bf [doc] how to use Meshroom without building AliceVision (#1487)
73da16d7 Merge pull request #1520 from alicevision/doc-citation
c0737bbb [doc] readme: add citation
500dfd96 [ui] Change from i++ to ++i
ee910b96 [ui] Change buttons default display before reconstruction
56d91bde [ui] Change intrinsic  structure and key getter for table view
83a404a7 [ui] Add clipping to columns
4fb719e2 [ui] Update tableview to stop flicking at bounds
0876b9e3 [ui] Add vertical scrollbar for high number of intrinsics
eecdcffb [ui] Enable focus change when clicking outside of text inputs
6bb5f84f [ui] Clean forgotten debug
6f5b6590 [ui] Adjust column width for serial number
7c0a9666 [ui] Clean code
b0254de1 [ui] UI adjustements for intrinsic display
79955dda [core] Change attribute name and label properties
1a506f81 Merge pull request #1509 from alicevision/dev/offsetFromCenter
1704e893 [core] fix upgrade for linked ListAttribute
61308eb2 [nodes] CameraInit: Principal Point is now relative to the image center
fdfd2e7b [nodes] Texturing: add refMesh and Normal / Height Map file type
2eb8783c [ui] Change Intrinsic display QML structure
7a8f78f5 [ui] Fix combo box height and placeholder display status
713b2f22 [ui] Add the possibility to dynamically change attribute values
776cbe78 [ui] Add combo box and check box for specific intrinsics
4cabfb79 [ui] Add dynamic view update on intrinsic changed
683633f1 [ui] Update intrinsic display
c1939e4e [ui] Add horizontal scrollbar to scroll across all intrinsics attributes
b0ee91dd [ui] Remove old display method files
379be4ad [ui] Update intrinsic parsing and displayed values
0deb9b49 [wip] Update intrinsic parsing
88cd6fb8 [wip] Add table view to display intrinsics infos
212d8178 [wip] Add function to build an object from intrinsics
a3c395fa [ui] Avoid undefined warning for intrinsic count
469f91c4 [ui] Fix scroll bar conflict
fef50023 [ui] Update enabled and visible states for intrinsics display
83c96000 [wip] Add intrinsic display inside image gallery
53b371df [wip] Display as list view
70c9f8b7 [wip] Add display of intrinsics
32c6d89a [ui] Add some padding
817c5dda [ui] Add number of intrinsics to display
f532ad76 [ui] Add filters to Image Gallery

git-subtree-dir: third_party/meshroom
git-subtree-split: c3cfa21b2521b893606ede7803c445fc97028a6c

michalpolic committed:
Merge commit Meshroom dev into dev

# Conflicts:
#	third_party/meshroom/start.sh

Commit:
Update path to Meshroom

Commit:
Hide 3D Viewer in default

michalpolic committed:
Add HoloLensMatcher parameters into the node

Add clustering radius and matching threshold into HoloLensMatcher node.
Add initialization of example dense mapping pipeline.

Commit:
Merge branch 'dev' of github.com:michalpolic/hololens_mapper into dev

Commit:
Update the HoloLens Matcher

Update paths for working with containers. Include Meshroom dependences.

michalpolic committed:
Update basic keyframe selector

Add support for selecting keyframes in tracking images.
Adjust and save related csv files.
Add copying of the depthmaps.

michalpolic committed:
Init of HoloLensIO node

Specify the input params of the node.

michalpolic committed:
Update init.sh

Add script for building conda enviroment.

michalpolic committed:
Fix simpleFarm import error

michalpolic committed:
Update init.sh to compile Cpp codes

michalpolic committed:
Update visibility estimation function and naming

Upadate the visibility estimation, i.e., hash points and estimate the
visibility only on s subsampled grid of points. Update naming of related
functions.

Commit:
Fix path to dense pointcloud model

Commit:
Fix cmake building in innit.sh

Commit:
Fix bugs of the VisibilityEstimation node

Change the transformation of the VLC cameras of HoloLens. Fix their
rotation matrices to have determinant equal one, update filtering of
visible points to  be in front of the camera. Update the camera params
in the hloc_map_composition.mg according new external clibration.

michalpolic committed:
Update instalation

Update instructions and init.sh

michalpolic committed:
Update the eviroment and cpp binding

michalpolic committed:
Update dense pointcloud output

michalpolic committed:
Add MatchingPairSelector node

Add MatchingPairSelector node into the Meshroom. Update paths in
previous nodes related to hloc map compositon.

michalpolic committed:
Update MeshroomCpp to suppot VS compiler

michalpolic committed:
Fix bug in R2q conversion

michalpolic committed:
Clear the codes, update gitignore

michalpolic committed:
Macher loading of all cams, preparation of workdir

Commit:
Fix bugs in HololensMapper

Update the HoloLensMapper to process all the images (add vlc images).
Mapper detect SIFT features, do a matching, then filter outliers far
from HoloLens tracking geometry.
Results are saved into COLMAP database.

Commit:
Init ColmapMapper node

michalpolic committed:
Update ColmapMapper

Runing version of the COLMAP mapper. The node takes COLMAP database
and compute SfM using COLMAP container.

michalpolic committed:
Init ModelsAligner node

Initialization of the node for aligning the COLMAP models. Update
output path of ColmapMapper.

Commit:
Finish ModelAligner node

Finished simple sparse mapping pipeline using Meshroom nodes.

Commit:
Squashed 'third_party/Hierarchical-Localization/' content from commit 775b6c1

git-subtree-dir: third_party/Hierarchical-Localization
git-subtree-split: 775b6c1bbb11656711495252afeb28808e4a69fa

Commit:
Merge commit '1a36a044734b1c42389359ebdb77805814e32a70' as 'third_party/Hierarchical-Localization'

Commit:
Add buildnig of HLOC container

Update Dockerfile, add Singularity.def and update the enviroment libs.

Commit:
Init HLOC node and wrapper

michalpolic committed:
Add HLOC map creator node into pipeline

Commit:
Remove the third_party submodules from HLOC

Commit:
Update HlocMapCreator

Add HlocMapCreator into the hloc_map_composition pipeline.
Update paths in Singularity definition file.

Commit:
Merge commit 'bbe8063c1d5d07bd7d6f5c5b052735958da4d8d1' as 'third_party/Hierarchical-Localization/third_party/d2net'

Commit:
Squashed 'third_party/Hierarchical-Localization/third_party/d2net/' content from commit 6b85495

git-subtree-dir: third_party/Hierarchical-Localization/third_party/d2net
git-subtree-split: 6b85495bbbde652f1cac5384d347db9437b1e6cf

Commit:
Add 'third_party/Hierarchical-Localization/third_party/SuperGluePretrainedNetwork/' from commit '7bb31186c119e9fceb6e8300fad3374b56f6e0aa'

git-subtree-dir: third_party/Hierarchical-Localization/third_party/SuperGluePretrainedNetwork
git-subtree-mainline: 3471ff5962e1feb0c3c1fca250fc60b485a6fd1a
git-subtree-split: 7bb31186c119e9fceb6e8300fad3374b56f6e0aa

Commit:
Merge commit 'c9650dd042b03e56e285e6f9a1b81f0819874500' as 'third_party/Hierarchical-Localization/third_party/deep-image-retrieval'

Commit:
Squashed 'third_party/Hierarchical-Localization/third_party/deep-image-retrieval/' content from commit 610247f

git-subtree-dir: third_party/Hierarchical-Localization/third_party/deep-image-retrieval
git-subtree-split: 610247f721452393738690fc56d0f66f5823d581

Commit:
Squashed 'third_party/Hierarchical-Localization/third_party/r2d2/' content from commit 0ff8f6a

git-subtree-dir: third_party/Hierarchical-Localization/third_party/r2d2
git-subtree-split: 0ff8f6afcbea91f19613d0cb7d93143a977830f5

Commit:
Merge commit 'fb89353acd3bc6fa59672fb99d70aeeff060c8d9' as 'third_party/Hierarchical-Localization/third_party/r2d2'

Commit:
Finished Hloc map creation codes

Add Net-VLAD model into HLOC third_party sources (download in inti.sh).
Update file seperator in images names.
Add conversion of COLMAP from TXT to BIN SfM representation.
Update gitignore to ignore NetVLAD model weights.

michalpolic committed:
Add converion COLMAP - Meshroom

michalpolic committed:
Update pilelines name

michalpolic committed:
Add the evaluation of commands in container

If node specify the 'containerName' the comand is rewritten and
run inside the specified container.

michalpolic committed:
Add SfM conversion to dense pipeline

Add node to convert Meshroom JSON to Meshroom ABC file.

Commit:
Update paths and running of the containers

Update the parameters to run any container just by its specification in
node definition.

michalpolic committed:
Add Meshroom nodes into dense mapping

Add nodes from Meshroom into dense mapping pipeline.
Adjust the nodes to run the container instead of executables.

michalpolic committed:
Fix the inputs in Meshroom node PrepareDenseScene

Dynamic size (original solution) is not compatabile with container
executable.

michalpolic committed:
Update HoloIO images paths

Update the conversion of SfM formats in HoloIO.
Images paths can be adujsted to run in the containers relative paths.

Commit:
Let the container specification as advanced param

The container prefix and name is setuped in nodes as additional
parameter.

Commit:
Merge branch 'dev' of github.com:michalpolic/hololens_mapper into dev

Commit:
Merge node paths 2

Commit:
Fix Meshing node outputs

michalpolic committed:
Init DensePointcloudAligner node

michalpolic committed:
Update dense mapping pipeline

Add all the required nodes to merge MVS with HoloLens depth sensor data.

michalpolic committed:
Update dense mapping pipeline

Smoth the output mesh before texturing

Commit:
QickFix of dense mapping pipeline nodes

Update images list to dictionary. Remove  parameters for texturing
from command line which are not supported by the container. Add update
of the points3D in HoloLensIO even for the COLMAP input.

Commit:
Merge branch 'dev' of github.com:michalpolic/hololens_mapper into dev

michalpolic committed:
Add filtering tracking cameras before texturing

Commit:
Change images list to images dictionary

michalpolic committed:
Update group for node container parameter

The code rewriting the container setting in cmd does not assume the
parameters in any other group than 'all_params'. Container parameters
should not be directly in cmd and therfore the group was chaged.

Commit:
Update COLMAP Mapper BA properties

Commit:
Update images list to dictionary

Commit:
Update HLOC map creation pipeline

The HoloLensIO node should not change the images paths. "imagesPath"
property should have value "original"

michalpolic committed:
Squashed 'third_party/AliceVision/' content from commit 55f40020c

git-subtree-dir: third_party/AliceVision
git-subtree-split: 55f40020c6402f47bc182501e901e988f2b7311d

michalpolic committed:
Merge commit '59f72301b73f04a3ac495100d62e41fe7b6bb888' as 'third_party/AliceVision'

michalpolic committed:
Update dense reconstruction pipeline

Try the JSON format as input for meshing.

michalpolic committed:
Merge branch 'dev' of github.com:michalpolic/hololens_mapper into dev

michalpolic committed:
Update dense reconstruction pipeline

Use the pointcloud from meshing to texturing.

Commit:
Update of the pipelines

Commit:
Squashed 'third_party/PatchmatchNet/' content from commit 82206d8

git-subtree-dir: third_party/PatchmatchNet
git-subtree-split: 82206d8b603ec925b6e4b1990618e0ad769347de

Commit:
Merge commit '8564245f5a2cc85011378ca456512e1a049e1146' as 'third_party/PatchmatchNet'

Commit:
Init PatchmatchNet Dockerfile

michalpolic committed:
Clear the repository

Remove old structures. The root directory should contain only pipelines,
tests and src.

michalpolic committed:
Add Dockerfile for PatchmatchNet

Add files to build PatchmatchNet container in Docker.
Add test to run simple depth estimation on small dataset.
TODO: add downloading of test data and set path in PatchmatchNet test

michalpolic committed:
Add new pipeline with PatchmatchNet

Init the nodes and pipeline in the form to be able run PatchmatchNet.
TODO: finish the code in PatchmatchNet node

Commit:
Add PatchmatchNet - Singularity definition

Add container definition for PatchmatchNet into Singularity.
Add COLMAP undistortion.
Update the PatchmatchNet node - runs in standard setting.

Commit:
Add reading binary PLY files

michalpolic committed:
Update PatchmatchNet Singularity definition

Update the Singularity definition file to work with RTX 3090 ti.
TODO: the home directory in Singularity definition is hardcoded,
generate it in init.sh before runing defintion file

michalpolic committed:
Merge branch 'dev' of github.com:michalpolic/hololens_mapper into dev

mpol007 committed:
Update eviroment for cloud usage

michalpolic committed:
Update eviroment for cloud usage

michalpolic committed:
Comment the codes related to SuperGlue, Patch2Pix

These codes has to be moved into containers otherwise we have too
complicated env that it does not run on CIIRC Cluster.

michalpolic committed:
CIIRC cluster update of enviroments

Commit:
Add multiprocessing for PatchmatchNet preparation

Evaluate the score of images to be used in MVS in parallel.

michalpolic committed:
Add caching of PatchmatchNet results

michalpolic committed:
Speed up the depthmap fusion

Add parallel prcessing of the depth verification and fusion.

michalpolic committed:
Optimize memory usage for HoloLensIO node

michalpolic committed:
Add pointcloud coloring in HoloLensIO node

Add coloring and export of pointcloud in HoloLensIO node. Change the
images list into the dictionary in Matcher related nodes.

michalpolic committed:
Update PatchmatchNet singularity definition file

TODO: it is still  specific for given PC, we have to generalize it.

michalpolic committed:
Update the HLOC map creation pipeline

The HoloLensIO has been updated. Update the parametes.

michalpolic committed:
Add pipeline for processing multiple recordings

michalpolic committed:
Update PointcloudComposer for HoloLens2

Add support of multiple data sources. Implement the code for composing
single pointcloud from HoloLens2 recording.

michalpolic committed:
Add reindexing cameras in model aligner

The Patchmatchnet requires consequtive indexing for cameras. Update
these indexes in ModelAligner node which is the last one before
Patchmatchnet.

michalpolic committed:
Add check for image id

Add check in composing view graph if the image id mentioned in point 3D
track does exist.

michalpolic committed:
Update the view graph

Update the ids in the viewgraph to be independent on image ids.
Image ids may have any values but we use their order id in viewgraph.

michalpolic committed:
Update KeyframeSelector to support HoloLens2

michalpolic committed:
Update IO convertor to work with HoloLens2

michalpolic committed:
Add dense mapping pipeline for HoloLens2

michalpolic committed:
Add HoloLens1Downloader

Add node for downloading the data from HoloLens 1.

michalpolic committed:
Init node for HoloLens2Downloader

TODO: add the functionality for HoloLens2Downloader.

michalpolic committed:
Add HoloLens2ForCV

Commit:
upload new nodes

Commit:
Update init for RICP

michalpolic committed:
Add HoloLens2Downloader

Add the node for downloading the HoloLens2 recordings.
TODO: add HoloLens2ForCV subtree

michalpolic committed:
Add manually HoloLens2ForCV

The subtree adding does not work. The codes were added manually as
no modifications of this codes in online repository are expected.

michalpolic committed:
Update the imput preprocessing nodes

Update the preprocessing nodes to follow the general scheme.
Rename HoloLensIO to IOConvertor.

michalpolic committed:
Inititalize the KeypointsDetector node

michalpolic committed:
Add HoloLens 2 calibration into IO Convertor node

michalpolic committed:
Add TentativeMatcher node

Add the node to find tentative matches and prefilter them by tracking
geometry constrains. Update the naming.

michalpolic committed:
Merge remote-tracking branch 'origin/zderaann' into dev_update_pipeline_nodes

zderaann committed:
Add files via upload

michalpolic committed:
Add conversion to JPG

Update naming and add conversion to JPG.

michalpolic committed:
Merge remote-tracking branch 'origin/zderaann' into dev_update_pipeline_nodes

michalpolic committed:
Fix. bugs in run_hloc.py

Update the paths and naming for all the functions.

michalpolic committed:
Add similar images from map to local SfM

This approach select similar images to local SfM images in global map.
Filter images with small number of correspondences and try to register
found images into the local SfM.
This approch does not work because of low accuracy of the localization.
TODO: add the generalized absolute pose solver and localize the whole
local SfM w.r.t. the global map.

michalpolic committed:
Merge commit 'fad8e66c14285a3408f7cf68ff271da38242b096' as 'third_party/poselib'

michalpolic committed:
Squashed 'third_party/poselib/' content from commit 35cf8989d

git-subtree-dir: third_party/poselib
git-subtree-split: 35cf8989ddbe721209e8d314eaa94447ab2bd504

michalpolic committed:
Add notes to install the poselib

michalpolic committed:
Add generalized absolute pose localization

Add the localization by generalized absolute pose solver to localize
sequence of images reconstructred in local SfM.

michalpolic committed:
Update HlocLocalizer node

Add merging of the global and local SfM based on the generalized
absotute pose solver, i.e., one pose for the whole local reconstruction
in global map.

michalpolic committed:
Update singularity enviroments (HLOC, POSELIB)

This does not lead to working version. Commit the current progress on
these definition files.

michalpolic committed:
Squashed 'third_party/pixel-perfect-sfm/' content from commit 980953c

git-subtree-dir: third_party/pixel-perfect-sfm
git-subtree-split: 980953c590173e19d505461d478c8991e41dfc6c

michalpolic committed:
Merge commit '2296e0a26922b589ec218260375e7b19ba18aa7a' as 'third_party/pixel-perfect-sfm'

michalpolic committed:
Update poselib and HLOC definition files

The compilation was working well for poselib. The same code part is in
HLOC. However,  both scripts fail now because of GPG error - n
ot available PUBKEY for Nvidia package.
May be caused by error on Nvidia side. TODO: check it

michalpolic committed:
Update Singularity .def files  (HLOC, Poselib)

Fix the bug caused by Nvidia GPT key change for the packages.
The containers HLOC and Poselib are compiled in Singularity without
errors.

michalpolic committed:
Update hloc container definition

Update the paths used inside the container. Update the enviroment
names to not match the local enviroments used on the host machine.

michalpolic committed:
Finish dense mapping pipeline

Fix bug in keyframe selector. Add nodes into the dense mapping pipeline
to merge several dense reconstructions.

michalpolic committed:
Create map creator pipeline

Create map creator pipelne and update Singularity definition files for
HLOC and Poselib (fix the missing file - change done in Nvidia images).

Commit:
Remove old pipelines

michalpolic committed:
Udate HLOC Singularity def file

Update the linked libraries to run pytorch on GPU RTX3090.

michalpolic committed:
Merge branch 'dev_update_pipeline_nodes' of github.com:michalpolic/hololens_mapper into dev_update_pipeline_nodes

Commit:
Update Singularity def files to follow HLOC one

The HLOC dependences works with GPU RTX3090. Copy the common part of the
dependences into the other Singularity def files.

michalpolic committed:
Update the pipelines to support new nodes

michalpolic committed:
Add depth cam filtering for HoloLens 2

Add min/max depth filter for ToF depth images.

michalpolic committed:
Fix naming for dense pts aligning

Rename the dense point cloud aligner and concatenator to fit its names.
The names were switched.

michalpolic committed:
Fix loading of OBJ point clouds

The input points can have colors in units 0-255 or 0-1.
Add support for both formats. Originally the code supported only 0-255.

michalpolic committed:
Add dense pts cloning: merge map pts and local pts

This update allows copying the dense point cloud of the map through the
dense mapping pipeline to be able merge it with the local dense
reconstruction.

michalpolic committed:
Add depthmaps and hashed pts to IOConvertor output

Add support to save the depthmaps and hashed point clouds generated
internally by IOConvertor.

michalpolic committed:
Update Pixel-Perfect-SfM signularity definition

The container use in runtime wrong MKL library from hosting OS.
This leads to falls. TODO: Fix this bug.

michalpolic committed:
Add rendering scale to IOConvertor

This is not so much usefull because the noise appears by the
up and downscaling. The better approach would be to project directly
the grid coorners into the image. TODO: update the rendering from points
to projections of the voxels.

michalpolic committed:
Merge pull request #3 from michalpolic/dev_update_pipeline_nodes

Dev update pipeline nodes
